### ISSUE TITLE
Time conversion and upper scale text lose contrast when changing photon wavelength

### ISSUE THREAD

**Issue by vperezcadavid99** [phetsims/quantum-wave-interference#486]
**Test device**
Chromebook

**Operating System**
ChromeOS

Browser
Chrome

**Problem description**
From https://github.com/phetsims/qa/issues/1430

This is a particular case where the color contrast test fails. I’m not sure if it is relevant, but I’m reporting it anyway. 

During the High Intensity screen, under the Photons particle configuration and Amplitude Wave Display, the time conversion and the upper scale text may become difficult to see when changing the photons' wavelength.

**Visuals**

<img width="2192" height="1056" alt="Image" src="https://github.com/user-attachments/assets/266816ed-1f5b-4989-9138-72d3a9396ff9" />

<details>
Troubleshooting information:
!!!!! DO NOT EDIT !!!!!
Name: ‪Quantum Wave Interference‬
URL: https://phet-dev.colorado.edu/html/quantum-wave-interference/1.0.0-rc.2/phet/quantum-wave-interference_all_phet.html
Version: 1.0.0-rc.2 2026-08-20 21:08:52 UTC
Features missing: applicationcache, applicationcache, touch
Flags: pixelRatioScaling
User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/151.0.0.0 Safari/537.36
Language: en-US
Window: 1600x812
Pixel Ratio: 1.7999999523162842/1
WebGL: WebGL 1.0 (OpenGL ES 2.0 Chromium)
GLSL: WebGL GLSL ES 1.0 (OpenGL ES GLSL ES 1.0 Chromium)
Vendor: WebKit (WebKit WebGL)
Vertex: attribs: 16 varying: 30 uniform: 4096
Texture: size: 16384 imageUnits: 16 (vertex: 16, combined: 32)
Max viewport: 32767x32767
OES_texture_float: true
Dependencies JSON: {}
<details/>

**Comment by samreid** [comment]
@matthew-blackman want to make the readouts black in that case, or some other solution?

**Comment by matthew-blackman** [comment]
Thanks for calling this out @vperezcadavid99. I went ahead and implemented the fix that you me and @samreid discussed earlier.

The distance scale marker and localized label are now contained in a borderless, semi-transparent black panel anchored to the wave area’s upper-left corner. The time conversion is contained in a matching panel anchored to the lower-left corner. Both panels automatically resize with their localized content while preserving their specified anchors, and they are non-pickable so they do not interfere with wave-area input.

@samreid, could you please review these code changes and, if everything looks good, reassign this issue to QA for review?

**Comment by samreid** [comment]
Code and runtime look great to me, I am just seeing the smallest 1px imperfection where at some screen resolutions the box sticks outside the wave area slightly:

<img width="496" height="356" alt="Image" src="https://github.com/user-attachments/assets/5415e02a-a7ed-4306-af48-b6e7442d867b" />

Worth fixing? Only you can decide

**Comment by matthew-blackman** [comment]
I feel that a fix is warranted as long as it is very straightforward and quick. I am having trouble reproducing the behavior on my display, but @samreid offered to take a look with his external monitor and make the 1px adjustment. Thanks @samreid!

**Comment by samreid** [comment]
I fixed the overlap in the commit and tested on retina and non-retina. 

However, I realized there is a downside to this proposed approach in that: when the wave passes those rectangles, it appears to "fan out". The propagative wave gets 15% wider or so when passing those rectangles which may conflate somewhat with the "fan out" after the barrier. So I'm interested in looking for another way that doesn't have this disadvantage.  I think a user using the sim for a while would understand this, but on a first glance, it seems like the first "physical" thing to happen after the wave emits from the left.

**Comment by samreid** [comment]
In the commit, I only show the background when necessary. This leaves the default configuration (red) and many other colors very clear and readable (with no "fan out" problem) while showing the background on colors with an otherwise lower contrast ratio. @matthew-blackman can you please test and advise?

**Comment by matthew-blackman** [comment]
Thanks @samreid! This change is working well in my testing for all of the photon wavelengths I tried.

I do have a couple of concerns, though. I’m not completely convinced that the “fanning out” behavior from the background panels was a big enough problem to warranting this conditional behavior. To me, it was still visually clear what was happening and didn’t seem likely to be confusing for students, though that is of course just my impression.

I’m also a little concerned about the “sometimes show the background” approach. I wasn’t able to exhaustively test every photon color, so I’m not certain that all configurations will pass contrast. And if the fanning-out effect is a concern, showing the background sometimes means that effect will still appear sometimes, which seems a little inconsistent as a solution.

More importantly, I noticed that while this works well for the photon colors I tested, the Matter Wave scenes would also need the background panel color. See screenshot:

<img width="523" height="351" alt="Image" src="https://github.com/user-attachments/assets/45c21354-cba4-4acf-b6fd-c60d3978dcdf" />

For this issue, I see a few possible paths:

1. Reduce the height of the background panels and the margins between the labels and the edges of the wave area, minimizing the fanning-out effect (I did this in the above commit).
2. Remove the conditional behavior and use the background consistently if the reduced size panels minimized the effect enough sufficiently.
3. Keep the conditional-background approach, but expand it to all Matter Wave scenes and more thoroughly test all photon colors.

My inclination is that with the smaller panels, we may be able to remove the conditional behavior you added and just use the backgrounds consistently. But I’m curious what you think after looking at the updated panel size.

Back to you for design feedback @samreid.

**Comment by samreid** [comment]
I feel the smaller size reduces the magnitude of the problem, but as I look at it, I still feel the part where the wave passes the rectangle is too "attention grabbing". That being said, I may be overthinking it, and I will ultimately trust your design instincts on any of the ways you wish to proceed for this issue.

**Comment by matthew-blackman** [comment]
Thanks, @samreid. I updated the scale panels so their backgrounds are now visible for all 'matter wave' scenes. I also reduced the panel opacity by 20%, including the dynamic opacity used for photons. I tested all particle types on both affected screens, and the labels still passed contrast checks in every case I found.

I’m reassigning this issue to @vperezcadavid99 for verification. Valentina, if everything looks good, please mark it for cherry-picking. Thanks, all!

**Comment by vperezcadavid99** [comment]
I'm seeing that the Normal text fails, but the Large text passes in this case. It still needs some changes.

<img width="502" height="306" alt="Image" src="https://github.com/user-attachments/assets/13d8c249-9688-4899-80f1-145b6d04baac" />

**Comment by matthew-blackman** [comment]
Thanks @vperezcadavid99! This has been fixed in the commit. Please verify, thanks!

**Comment by vperezcadavid99** [comment]
I noticed a couple of regions that are still failing the contrast check (highlighted in red, while the green ones pass). Could we apply a shaded background to these areas as well? For these specific regions, we would need a lighter shade using black text instead of a darker one.

<img width="192" height="110" alt="Image" src="https://github.com/user-attachments/assets/d86abd6c-1c5b-45c4-8960-ad0a510f29c7" />

**Comment by samreid** [comment]
On slack, @matthew-blackman mentioned considering moving these readouts outside of the wave area. Self-unassigning while that is under consideration.

**Comment by matthew-blackman** [comment]
As @samreid said, I’m considering a simpler approach for this issue: moving the distance scale and time-conversion text outside the wave area on both the High Intensity and Single Particles screens, as shown in the High Intensity mockup:

<img width="1436" height="930" alt="Image" src="https://github.com/user-attachments/assets/18570e4b-d410-4f33-83da-7eb12d7ecd18" />

This would provide consistent contrast without requiring background panels or wavelength-dependent opacity and color logic, reducing the visual and implementation complexity. I'd like to review this with @arouinfar before implementing.

**Comment by matthew-blackman** [comment]
I met with @arouinfar to review the proposed mockup and discuss moving the scale readouts outside the wave area. She agreed that this is a clearer and simpler solution than the background-panel approach and liked the proposed placement.

We’re proceeding with the time scale below the wave area, near its left edge with a small horizontal and vertical margin. The distance scale will appear above the wave area, inset to the right on High Intensity as shown in the mockup, and left-aligned with the time scale on Single Particles. This eliminates the panels and wavelength-dependent contrast logic.

**Comment by matthew-blackman** [comment]
I replaced the contrast panels and wavelength-dependent styling with a simpler layout. The distance and time scale readouts now use black content and are positioned outside the wave area on the High Intensity and Single Particles screens, with screen-specific alignment.

Thanks, everyone! @samreid, do you mind checking the commit to make sure all of the previous behavior was replaced? If so, please mark as ready-to-cherry-pick.

**Comment by samreid** [comment]
Code review is good, nice work

**Comment by vperezcadavid99** [comment]
Nice solution for this! Looks good.
Closing
